docs/installation.rst

```diff
@@ -1,11 +1,46 @@
 Installation
 ============
 
-To install this package run
+To use StarkNet.py, ``ecdsa, fastecdsa, sympy`` dependencies are required. Depending on the operating system,
+different install steps must be performed.
 
-``pip install starknet.py``
+Linux
+^^^^^
 
-or using Poetry:
+.. code-block:: console
 
-``poetry add starknet.py``
+    sudo apt install -y libgmp3-dev
+    pip install starknet-py
 
+MacOS with Intel processor
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires `Homebrew <https://brew.sh/>`_  installation.
+
+.. code-block:: console
+
+    brew install gmp
+    pip install starknet-py
+
+MacOS with Apple silicon
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires `Homebrew <https://brew.sh/>`_  installation.
+First install gmp:
+
+.. code-block:: console
+
+    brew install gmp
+
+Then install required dependencies and StarkNet.py itself:
+
+.. code-block:: console
+
+    CFLAGS=-I`brew --prefix gmp`/include LDFLAGS=-L`brew --prefix gmp`/lib pip install ecdsa fastecdsa sympy
```

Comment:
cmake is also needed for fastecdsa

Reply:
Are you sure? And is it only on Apple silicon macs?

Reply:
Kuba has been installing it yesterday and was missing cmake. Not sure if it is only needed for ARM ones though.

Reply:
I think we can add it to both versions just to be sure

Reply:
It might be that the problem is caused by `wheel` not being installed causing pip to fall back to old installation method. I'm not really sure if I had to install cmake for fastecdsa to work.

Reply:
Created #600 for this. Merging the current version.